url433bec1e56_sarobe_DifficultyInvestigation_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/fr/inria/optimization/cmaes/CMAEvolutionStrategy.java

```diff
@@ -0,0 +1,2704 @@
+        }
+        
+        /** @return Maximum value of 1-D double array */
+        public double max(double ar[]) {
+            int i;
+            double m;
+            m = ar[0];
+            for (i = 1; i < ar.length; ++i) {
+                if (m < ar[i])
+                    m = ar[i];
+            }
+            return m;
+        }
+
+        /** sqrt(a^2 + b^2) without under/overflow. **/
+        public double hypot(double a, double b) {
+            double r  = 0;
+            if (Math.abs(a) > Math.abs(b)) {
+               r = b/a;
+               r = Math.abs(a)*Math.sqrt(1+r*r);
+            } else if (b != 0) {
+               r = a/b;
+               r = Math.abs(b)*Math.sqrt(1+r*r);
+            }
+            return r;
+         }
+        /** @return index of minium value of 1-D double array */
+        public int minidx(double ar[]) {
+            return minidx(ar, ar.length-1);
+        }
+        
+        /** @return index of minium value of 1-D double 
+         *   array between index 0 and maxidx 
+         * @param ar double[] 
+         * @param maxidx last index to be considered */
+        public int minidx(double[] ar, int maxidx) {
+            int i, idx;
+            idx = 0;
+            for (i = 1; i < maxidx; ++i) {
+                if (ar[idx] > ar[i])
+                    idx = i;
+            }
+            return idx;
+        }
+
+        /** @return index of minium value of 1-D double 
+         *   array between index 0 and maxidx 
+         * @param ar double[] 
+         * @param maxidx last index to be considered */
+        protected int minidx(IntDouble[] ar, int maxidx) {
+            int i, idx;
+            idx = 0;
+            for (i = 1; i < maxidx; ++i) {
+                if (ar[idx].val > ar[i].val)
+                    idx = i;
+            }
+            return idx;
+        }
+
+        /** @return index of maximum value of 1-D double array */
+        public int maxidx(double ar[]) {
+            int i, idx;
+            idx = 0;
+            for (i = 1; i < ar.length; ++i) {
+                if (ar[idx] < ar[i])
+                    idx = i;
+            }
+            return idx;
+        }
+        /** @return Minimum value of 1-D double array */
+        public double min(double ar[]) {
+            int i;
+            double m;
+            m = ar[0];
+            for (i = 1; i < ar.length; ++i) {
+                if (m > ar[i])
+                    m = ar[i];
+            }
+            return m;
+        }
+        
+        /** @return Maximum value of 1-D Object array where the object implements Comparator 
+         *    Example: max(Double arx, arx[0]) */
+        public Double max(Double ar[], Comparator<Double> c) {
+            int i;
+            Double m;
+            m = ar[0];
+            for (i = 1; i < ar.length; ++i) {
+                if (c.compare(m, ar[i]) > 0)
+                    m = ar[i];
+            }
+            return m;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 987:992, 1023:1028.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.